core/src/main/java/org/apache/iceberg/io/OutputFileFactory.java

```diff
@@ -143,12 +153,17 @@ public Builder format(FileFormat newFormat) {
       return this;
     }
 
+    public Builder suffix(String newSuffix) {
+      this.suffix = newSuffix;
+      return this;
+    }
+
     public OutputFileFactory build() {
       LocationProvider locations = table.locationProvider();
       FileIO io = table.io();
       EncryptionManager encryption = table.encryption();
       return new OutputFileFactory(
-          defaultSpec, format, locations, io, encryption, partitionId, taskId, operationId);
+          defaultSpec, format, locations, io, encryption, partitionId, taskId, operationId, suffix);
```

Comment:
Which tests caught the need for the suffix?

Reply:
most/all of the tests that extend `SparkRowLevelOperationsTestBase` were failing when reading data (after delete files have been written)